products/nativescript/tns_logs.py

```diff
@@ -197,6 +197,46 @@ def __app_refresh_messages(instrumented, app_type):
                 logs.append('QA: items component on init')
         return logs
 
+    @staticmethod
+    def __webpack_messages():
+        logs = []
+        logs.append('File change detected.')
+        logs.append('Starting incremental webpack compilation...')
+        logs.append('Webpack compilation complete.')
+        logs.append('Webpack build done!')
+        return logs
+
+    @staticmethod
+    def preview_initial_messages(platform, bundle=False, hmr=False, instrumented=False):
+        logs = ['Start sending initial files for platform {0}'.format(str(platform)),
+                'Successfully sent initial files for platform {0}'.format(str(platform))]
+
+        if bundle or hmr:
+            logs.extend(TnsLogs.__webpack_messages())
+        if instrumented:
+            logs.append('QA: Application started')
+        return logs
+
+    @staticmethod
+    def preview_file_changed_messages(platform, file_name, run_type=RunType.INCREMENTAL,
+                                      bundle=False, hmr=False, instrumented=False):
+        logs = ['Start syncing changes for platform {0}'.format(str(platform))]
+        if bundle or hmr:
+            logs.extend(TnsLogs.__webpack_messages())
+            logs.append(file_name)
+        else:
+            logs.append('Successfully synced {0} for platform {1}'.format(file_name, str(platform)))
+        if hmr:
+            logs.append('Successfully synced bundle.')
+            logs.append('hot-update.json')
+            logs.append('HMR: Checking for updates to the bundle with hmr hash')
```

Comment:
We already have those HMR messages in __file_changed_messages.
Ideally I would like to have those string is single place (if possible).

Reply:
Yes I also thought of tat  but it appeared that only one message of the group differs. If you are doing run the message is 'Successfully transferred bundle.' if you are doing preview the same message is 'Successfully synced bundle.' and i could not think of a way to differentiate if I am doing bundle or run in method which returns hmr messages in general.